### ISSUE TITLE
Maybe Style/StringConcatenation should be conservative by default

### ISSUE THREAD

**Issue by jaynetics** [rubocop/rubocop#11757]
as per https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation, `aggressive` mode is currently the default.

this is not ideal because it triggers false positives and people have to research it and change the mode.

- we hit a case with `URI#+`, which returns a URI and prevents double slashes
  - e.g. `uri + '/some_path'` always returns a valid URI
- `Pathname#+` behaves similarly and thus yields similar false positives
- custom implementations of `#+` are [also affected](https://github.com/rubocop/rubocop/issues/10685)

**Comment by koic** [comment]
This configuration was introduced at https://github.com/rubocop/rubocop/pull/9153 with `aggressive` as the default. There are #9144 arguments, but I think I can accept this proposal to default to `conservative`. This is because I prefer to be as safe as possible, but others may think differently, like some of #9144.
@rubocop/rubocop-core What do you think about it?

**Comment by bbatsov** [comment]
I still stand by my original opinion. That cop is next to useless with the conservative config. Likely we'll just have it disabled by default in RuboCop 2.0, as one of the main goals is to streamline the default configuration (profile) and have be more conservative in general.

**Comment by bbatsov** [comment]
See also #10726

**Comment by noraj** [comment]
`rubocop -A` broke my code when it's supposed to be safe.

A minimal example of how this breaks filepath:

```ruby
# frozen_string_literal: true

require 'pathname'

FILENAME = 'mime.types'

pn = Pathname.new('/etc')
fp = pn + 'nginx' + FILENAME

puts fp # /etc/nginx/mime.types
```

```
➜ rubocop -A pathname.rb
Inspecting 1 file
C

Offenses:

pathname.rb:8:6: C: [Corrected] Style/StringConcatenation: Prefer string interpolation to string concatenation.
fp = pn + 'nginx' + FILENAME
     ^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

```ruby
# frozen_string_literal: true

require 'pathname'

FILENAME = 'mime.types'

pn = Pathname.new('/etc')
fp = "#{pn}nginx#{FILENAME}"

puts fp # /etcnginxmime.types
```

**Comment by bquorning** [comment]
> `rubocop -A` broke my code when it's supposed to be safe.

`-A` isn’t safe:

```
❯ rubocop --help | grep \\-A
    -A, --autocorrect-all            Autocorrect offenses (safe and unsafe).
```

**Comment by noraj** [comment]
Right, I should have used `-a`

![image](https://github.com/rubocop/rubocop/assets/16578570/cf29ab10-4f86-4380-9d6c-fc1530480c68)

**Comment by bbatsov** [comment]
Reopened — still reproduces on master (1.88). With the default config, `uri + '/some_path'` autocorrects to `"#{uri}/some_path"` and `path + 'bar'` to `"#{path}bar"`, so `URI#+`/`Pathname#+` are still flagged out of the box.

**Comment by enmachs** [comment]
@bbatsov , is this good to be taken for me? As my first time issue take. I'd would like to work on a fix, and set it up in my local for reproduce it and test :D
